src/engine/bank05.asm

```diff
@@ -4196,66 +4464,1748 @@ Func_16695: ; 16695 (5:6695)
+.check_recoil
+	ld a, MOVE_FLAG1_ADDRESS | LOW_RECOIL_F
+	call CheckLoadedMoveFlag
+	jr c, .is_recoil
+	ld a, MOVE_FLAG1_ADDRESS | HIGH_RECOIL_F
+	call CheckLoadedMoveFlag
+	jp nc, .check_defending_can_ko
+.is_recoil
+	; sub from AI score number of damage counters
+	; that move deals to itself.
+	ld a, [wLoadedMoveUnknown1]
+	or a
+	jp z, .check_defending_can_ko
+	ld [wDamage], a
+	call ApplyDamageModifiers_DamageToSelf
+	ld a, e
+	call CalculateTensDigit
+	call SubFromAIScore
+	
+	push de
+	ld a, MOVE_FLAG1_ADDRESS | HIGH_RECOIL_F
+	call CheckLoadedMoveFlag
+	pop de
+	jr c, .high_recoil
+
+	; if LOW_RECOIL KOs self, decrease AI score
+	ld a, DUELVARS_ARENA_CARD_HP
+	call GetTurnDuelistVariable
+	cp e
+	jr c, .kos_self
+	jp nz, .check_defending_can_ko
+.kos_self
+	ld a, 10
+	call SubFromAIScore
+
+.high_recoil
+	; dismiss this move if no benched Pokémon
+	ld a, DUELVARS_NUMBER_OF_POKEMON_IN_PLAY_AREA
+	call GetTurnDuelistVariable
+	cp 2
+	jr c, .dismiss_high_recoil_move
+	; has benched Pokémon
+
+; here the AI handles high recoil moves differently
+; depending on what deck it's playing.
+	ld a, [wOpponentDeckID]
+	cp ROCK_CRUSHER_DECK_ID
+	jr z, .rock_crusher_deck
+	cp ZAPPING_SELFDESTRUCT_DECK_ID
+	jr z, .zapping_selfdestruct_deck
+	cp BOOM_BOOM_SELFDESTRUCT_DECK_ID
+	jr z, .encourage_high_recoil_move
+	; Boom Boom Selfdestruct deck always encourages
+	cp POWER_GENERATOR_DECK_ID
+	jr nz, .high_recoil_generic_checks
+	; Power Generator deck always dismisses
+
+.dismiss_high_recoil_move
+	xor a
+	ld [wAIScore], a
+	jp .done
+
+.encourage_high_recoil_move
+	ld a, 20
+	call AddToAIScore
+	jp .done
+
+; Zapping Selfdestruct deck only uses this move
+; if number of cards in deck >= 30 and
+; HP of active card is < half max HP.
+.zapping_selfdestruct_deck
+	ld a, DUELVARS_NUMBER_OF_CARDS_NOT_IN_DECK
+	call GetTurnDuelistVariable
+	cp 31
+	jr nc, .high_recoil_generic_checks
+	ld e, PLAY_AREA_ARENA
+	call GetCardDamage
+	sla a
+	cp c
+	jr c, .high_recoil_generic_checks
+	ld b, 0
+	ld a, DUELVARS_ARENA_CARD
+	call GetTurnDuelistVariable
+	call GetCardIDFromDeckIndex
+	ld a, e
+	cp MAGNEMITE1
+	jr z, .magnemite1
+	ld b, 10 ; bench damage
+.magnemite1
+	ld a, 10
+	add b
+	ld b, a ; 20 bench damage if not Magnemite1
+
+; if this move causes player to win the duel by
+; knocking out own Pokémon, dismiss move.
+	ld a, 1 ; count active Pokémon as KO'd
+	call .check_if_kos_bench
+	jr c, .dismiss_high_recoil_move
+	jr .encourage_high_recoil_move
+
+; Rock Crusher Deck only uses this move if
+; prize count is below 4 and move wins (or potentially draws) the duel,
+; (i.e. at least gets KOs equal to prize cards left).
+.rock_crusher_deck
+	call CountPrizes
+	cp 4
+	jr nc, .dismiss_high_recoil_move
+	; prize count < 4
+	ld b, 20 ; damage dealt to bench
+	call SwapTurn
+	xor a
+	call .check_if_kos_bench
+	call SwapTurn
+	jr c, .encourage_high_recoil_move
+
+; generic checks for all other deck IDs.
+; encourage move if it wins (or potentially draws) the duel,
```

Comment:
this and the next line confuse me a bit. Will it encourage if it 

Wins or leads directly to a draw, or
Puts the AI in a better or equal state (aka <= prizes left compared to player)